### DISCUSSION TITLE
Mastodon behind reverse proxy

### DISCUSSION THREAD

**Discussion by ghost** [mastodon/mastodon#18360] [Q&A]
Hello,
I'm trying to set up my first ever mastodon server, and trying to put it behind a reverse proxy. I have a domain specifically for mastodon, with caddy as a proxy in front of it and a mastodon VM. Currently, I'm just using the standard reverse_proxy directive in the caddyfile for the domain I'm using. However, requests aren't being proxied to nginx. If I try and access the VMs local IP, it just returns four random characters in a string through http/0.9. I'm honestly entirely clueless where to start, but I'm perfectly ready to answer any questions.

**Comment by ClearlyClaire** [comment]
It might be an issue with HTTP/2 support in either Caddy or nginx, I would disable HTTP/2 entirely between Caddy and nginx.
